### ISSUE TITLE
[BUG] FIELD location not being handled

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#622]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

latest as of issue creation

### Priority

Critical

### Current Behavior

The following condition:
```yaml
  when:
    java.referenced:
      pattern: java.lang.String
      location: FIELD
  ```
does not match against
```java
public Class Person {
  private String name;
}
```

### Expected Behavior

The `FIELD` location should be available and work, so that we can match against fields of a given type.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Create a custom rule with the given condition
2. Have a `String` field to be matched against in the application to be analyzed, within any class
3. Run an analysis
4. See how no matches are found

### Anything else?
:exclamation: `FIELD` and `FIELD_DECLARATION` are the same. `FIELD_DECLARATION` was used in windup, while it looks like jdt.ls [uses `FIELD` instead](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/search/org/eclipse/jdt/core/search/IJavaSearchConstants.java#L72).
